src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java

```diff
@@ -87,4 +102,74 @@ public static String getNodeSelectorFromNodeIndentifier (NodeInfo node) {
       node.getNodeType() + "-" +
       node.getNodeId() + "\"]";
   }
+
+  public static void deployAndTestPipeline(String filename, String pipelineName) {
+    SeleniumDriver.openPage(Constants.PIPELINES_URL);
+    WaitHelper.waitForPageToLoad();
+    ElementHelper.clickOnElement(locateElementById("resource-center-btn"));
+    ElementHelper.clickOnElement(locateElementById("create-pipeline-link"));
+    if(!SeleniumDriver.getDriver().getCurrentUrl().contains("studio")) {
+      throw new RuntimeException("URL redirection for pipeline creation unsuccessful.");
+    }
+    WaitHelper.waitForPageToLoad();
+    WebElement upload_file = SeleniumDriver.getDriver()
+            .findElement(By.xpath("//*[@id='pipeline-import-config-link']/input[@type='file']"));
+
+    File pipelineJSONFile = new File("src/e2e-test/fixtures/" + filename);
+    String filePath = pipelineJSONFile.getAbsolutePath();
+    upload_file.sendKeys(filePath);
+
+//    ElementHelper.clickOnElement(
+//            WaitHelper.waitForElementToBeClickable(locateElementByCssSelector("div[class*='pipeline-name']"))
+//    );
+    // Below is the workaround as above(commented block) is not working
+    try {
+      WebElement pipelineNameContainer = locateElementByCssSelector("div[class*='pipeline-name']");
+      ElementHelper.clickOnElement(pipelineNameContainer);
+    } catch(org.openqa.selenium.StaleElementReferenceException ex) {
+      WebElement pipelineNameContainer = locateElementByCssSelector("div[class*='pipeline-name']");
```

Comment:
This also feels weird as you are the same steps in the try block again in the catch block (what if it fails again?)

Reply:
Found alternative and fixed it